src/coreclr/tools/Common/Compiler/TypeExtensions.cs

```diff
@@ -594,6 +602,16 @@ public static MethodDesc TryResolveConstraintMethodApprox(this TypeDesc constrai
                 method = method.MakeInstantiatedMethod(methodInstantiation);
             }
 
+            // It's difficult to discern what runtime determined form the interface method
+            // is on later so fail the resolution if this would be that.
+            // This is pretty conservative and can be narrowed down.
```

Comment:
I went down this rabbit hole and it lead to stuff like:

https://github.com/MichalStrehovsky/runtime/blob/c7e5c934159fc27b5c92a0c609a6756cf43a26fe/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Compilation.cs#L519-L622

I really didn't like it and had trouble getting it to work. The problem is that RyuJIT doesn't operate on runtime-determined things and we can't even properly resolve such constrained calls (the support for that was never needed in the type system because only universal shared code needs to resolve constrained calls on runtime determined types otherwise and that was always supported by the STS type system in NUTC).